awx/main/wsrelay.py

```diff
@@ -71,7 +73,16 @@ async def connect(self):
         if not self.channel_layer:
             self.channel_layer = get_channel_layer()
 
-        uri = f"{self.protocol}://{self.remote_host}:{self.remote_port}/websocket/relay/"
+        # figure out if what we have is an ipaddress, IPv6 Addresses must have brackets added for uri
+        uri_hostname = self.remote_host
+        try:
+            # Throws ValueError if self.remote_host is a hostname like example.com, not an IPv4 or IPv6 ip address
+            if type(ipaddress.ip_address(host)) == ipaddress.IPv6Address:
```

Comment:
I think `isinstance` is more conventional?

```suggestion
            if isinstance(ipaddress.ip_address(host), ipaddress.IPv6Address):
```

Reply:
oh, and:
```suggestion
            if isinstance(ipaddress.ip_address(uri_hostname), ipaddress.IPv6Address):
```

Reply:
good eye, will fix